Sources/HttpPipeline/Middleware.swift

```diff
@@ -15,90 +15,86 @@ public func writeStatus<A>(_ status: Status) -> Middleware<StatusLineOpen, Heade
   }
 }
 
-public func writeHeader<A>(_ header: ResponseHeader) -> Middleware<HeadersOpen, HeadersOpen, A, A> {
+public func writeHeader<E, A>(_ header: ResponseHeader) -> Middleware<HeadersOpen, HeadersOpen, E, E, A, A> {
   return pure <<< (\.response.headers %~ { $0 + [header] })
 }
 
-public func writeHeaders<A>(_ headers: [ResponseHeader]) -> Middleware<HeadersOpen, HeadersOpen, A, A> {
+public func writeHeaders<E, A>(_ headers: [ResponseHeader]) -> Middleware<HeadersOpen, HeadersOpen, E, E, A, A> {
   return pure <<< (\.response.headers %~ { $0 + headers })
 }
 
-public func writeHeader<A>(_ name: String, _ value: String) -> Middleware<HeadersOpen, HeadersOpen, A, A> {
+public func writeHeader<E, A>(_ name: String, _ value: String) -> Middleware<HeadersOpen, HeadersOpen, E, E, A, A> {
   return writeHeader(.other(name, value))
 }
 
-public func closeHeaders<A>(conn: Conn<HeadersOpen, A>) -> IO<Conn<BodyOpen, A>> {
+public func closeHeaders<E, A>(conn: Conn<HeadersOpen, E, A>) -> IO<Conn<BodyOpen, E, A>> {
   return pure <| .init(
     data: conn.data,
     request: conn.request,
     response: conn.response
   )
 }
 
-public func end(conn: Conn<BodyOpen, Data>) -> IO<Conn<ResponseEnded, Data>> {
-  return pure <| .init(
-    data: conn.data,
-    request: conn.request,
-    response: Response(
-      status: conn.response.status,
-      headers: conn.response.headers,
-      body: conn.data
+public func end<E>(conn: Conn<BodyOpen, E, Data>) -> IO<Conn<ResponseEnded, E, Data>> {
```

Comment:
This gets a lil complicated. There's definitely ways to clean up with some functions, but wonder if we can clean it up as is?